src/chain.h

```diff
@@ -339,6 +352,14 @@ class CDiskBlockIndex : public CBlockIndex
         READWRITE(nBits);
         READWRITE(nNonce);
         READWRITE(nSolution);
+
+        // Only read/write nSproutValue if the client version used to create
+        // this index was storing them.
+        // TODO: See if we can get away with not serializing the boost::optional
+        // one-byte header, without requiring users to reindex on upgrade.
+        if (nType & SER_DISK && nVersion >= SPROUT_VALUE_VERSION) {
```

Comment:
still don't understand the logic of this line. In particular, what do this variables on the LHS have to do with the version the client is running?

Reply:
It's an artifact of the way Bitcoin does serialization. `nType` can be one of `SER_NETWORK`, `SER_DISK` or `SER_GETHASH` (the latter used in very specific cases), By convention, if `nType == SER_NETWORK` then `nVersion` is set to the protocol version (because that serialization is for communicating with peers), while if `nType == SER_DISK` then `nVersion` is set to the client version (because that serialization is only read by the local client, often across local version upgrades).

Reply:
Thanks, I kind of get it. Is there a reason for the choice `1001400`?

Reply:
I wonder if it's worth adding parentheses for each of the two conditions; initially I parsed it as one condition.

Reply:
`1001400` is what `CLIENT_VERSION` will be set to from 1.0.14-rc1, which will be the first tagged "release" to have this code enabled.

Reply:
I agree with @arielgabizon parentheses would make it clearer, even though operator precedence is fine. re: http://en.cppreference.com/w/cpp/language/operator_precedence